spec/arbre/unit/context_spec.rb

```diff
@@ -5,32 +5,33 @@
 describe Arbre::Context do
 
   let(:context) do
-    Arbre::Context.new do
+    described_class.new do
       h1 "札幌市北区" # Add some HTML to the context
     end
   end
 
-  it "should not increment the indent_level" do
+  it "does not increment the indent_level" do
     expect(context.indent_level).to eq(-1)
   end
 
-  it "should return a bytesize" do
+  it "returns a bytesize" do
     expect(context.bytesize).to eq(25)
   end
 
-  it "should return a length" do
+  it "returns a length" do
     expect(context.length).to eq(25)
   end
 
-  it "should delegate missing methods to the html string" do
+  it "delegates missing methods to the html string" do
     expect(context).to respond_to(:index)
     expect(context.index('<')).to eq(0)
   end
 
-  it "should use a cached version of the HTML for method delegation" do
-    expect(context).to receive(:to_s).once.and_return("<h1>札幌市北区</h1>")
+  it "uses a cached version of the HTML for method delegation" do
+    allow(context).to receive(:to_s).and_return("<h1>札幌市北区</h1>")
```

Comment:
I'm not a fan of using `allow` because it has two very different outcomes which can lead to question marks and can introduce false positives. I've been bitten by that enough. If this is critical to what's being tested, then it should use `expect` instead which by definition is clearer. In this case, based on the asserts, it seems it is.

There are multiple instances of this here so I won't comment on each one but all should be addressed.